engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -467,7 +407,8 @@ public boolean isActiveEntity(long id) {
     @Override
     //Todo: implement iterating over multiple pools
     public Iterable<Component> iterateComponents(long entityId) {
-        return globalPool.getComponentStore().iterateComponents(entityId);
+        return Iterables.concat(globalPool.getComponentStore().iterateComponents(entityId),
+                sectorManager.getComponentStore().iterateComponents(entityId));
```

Comment:
This is basically an "either ... or ..." in case an entity is either in sector or global scope, is that correct? If, for some reason, an entity ends up to be in both, this will duplicate all the components, won't it?